anesthetic/examples/perfect_ns.py

```diff
@@ -5,7 +5,8 @@
 from anesthetic.samples import merge_nested_samples
 
 
-def gaussian(nlive, ndims, sigma=0.1, R=1, logLmin=-1e-2):
+def gaussian(nlive, ndims, sigma=0.1, R=1, logLmin=-1e-2, logLmax=0,
+             *args, **kwargs):
```

Comment:
`logLmax` is lacking a docstring entry.

Reply:
Thanks for your comment. But logLmax needs to be a number. I'm not sure what you mean I am sorry...

Reply:
The [docstring](https://www.programiz.com/python-programming/docstrings) underneath the function definition is the text that describes what the function does such that a new user can quickly inform themselves on the correct usage. Such a function docstring should include a list of (at least) all the explicit function arguments. By numpy convention (which is the one that we are using), all these function arguments get listed underneath a "Parameters" heading. In jupyter notebooks this docstring is easily accessible by pressing `shift+tab` inside a function.

Here, the docstring currently is:
```python
    """Perfect nested sampling run for a spherical Gaussian & prior.

    Up to normalisation this is identical to the example in John Skilling's
    original paper https://doi.org/10.1214/06-BA127 . Both spherical Gaussian
    width sigma and spherical uniform prior width R are centered on zero

    Parameters
    ----------
    nlive : int
        number of live points#

    ndims : int
        dimensionality of gaussian

    sigma : float
        width of gaussian likelihood

    R : float
        radius of gaussian prior

    logLmin : float
        loglikelihood at which to terminate

    The remaining arguments are passed to the
    :class:`anesthetic.samples.NestedSamples` constructor.

    Returns
    -------
    samples : :class:`anesthetic.samples.NestedSamples`
        Nested sampling run
    """
```

So this currently lists `nlive`, `ndims`, `sigma`, `R`, and `logLmin`, but it lacks a description of `logLmax`.

Reply:
Thank you very much for explaining! I've implemented your suggestion.

Reply:
Hmm, I was hoping for a bit more description...

Then again, the docstring for the other arguments is also very minimalistic... and possibly a bit misleading... Is this really a nested sampling run of a spherical Gaussian and prior? Or would it be more accurate to call it a nested sampling run of a spherical Gaussian likelihood from a spherical top-hat prior? In which case the docstring for `R` should be rather the radius of the spherical top-hat prior...

Anyhow, it's not right to ask you to clean all that up (unless you want to), but it would be nice if you could make the `logLmax` docstring a bit more descriptive. Something along the lines that, yes, this defines the maximum of the Gaussian likelihood, but in effect this only shifts the log-likelihood up and down by a constant. Is there a reason anyone should choose a value different from the default `logLmax=0`? What prompted you to introduce this argument the first place? If that is a gotcha that would be good for other people to be aware of, then this is a good place to warn about it.

Reply:
Hi @lukashergt I am terribly sorry for my late reply, I somehow missed your newest comment. 

The reason why we need to add LogLmax is because we’ve found an issue with the definition of combined Gaussian likelihood. We are recently adding a new function (in the branch called tension) to anesthetic for calculating tension statistics between datasets. The initial test involves creating 2 mock datasets A and B with d-dimensional Gaussian likelihood and a combined dataset AB (combining A and B on the likelihood level). We discovered that the quantity of logLmax is actually needed explicitly in the calculation of combined likelihoods. Besides, we have noticed that the correlated_gaussian() function does not produce a normalised correlated gaussian likelihood and the evidence isn’t unity. Therefore, we need to update the function description. Please refer to equations 14-21 in the paper Quantifying tensions in cosmological parameters: Interpreting the DES evidence ratio [1902.04029]. Thank you very much!